ironmon_tracker/screens/LogOverlay.lua

```diff
@@ -915,71 +960,203 @@ function LogOverlay.buildPokemonZoomButtons(data)
 		offsetY = offsetY + Constants.SCREEN.LINESPACING
 	end
 
-	-- POKEMON ICON
-	local offsetEvoY = Utils.inlineIf(#data.p.evos > 0, -3, 0)
+
+	local evoMethods = Utils.getShortenedEvolutionsInfo(PokemonData.Pokemon[data.p.id].evolution) or {}
+
+	local preEvoList = {}
+	local evoList = {}
+
+	local hasPrevEvo = #data.p.prevos > 0 or false
+
+	if hasPrevEvo and Options["Show Pre Evolutions"] == true then
+		-- Add prevos to list
+		for i, prev in ipairs(data.p.prevos) do
+			table.insert(preEvoList, {
+				name = PokemonData.Pokemon[prev.id].name,
+				id = prev.id
+			})
+		end
+	else
+		hasPrevEvo = false
+	end
```

Comment:
Just noting a chance to do a little clean-up here if interested. You don't need to do an `local hasPrevEvo = [boolean] or false` since the [boolean] part will either be true, or false if false; so that's implied already. You can also merge the conditions that follow as such (I believe this is the same resulting outcome):

```lua
local hasPrevEvo = Options["Show Pre Evolutions"] and #data.p.prevos > 0
if hasPrevEvo then
	-- Add prevos to list
	for i, prev in ipairs(data.p.prevos) do
		table.insert(preEvoList, {
			name = PokemonData.Pokemon[prev.id].name,
			id = prev.id
		})
	end
end
```

Reply:
Went ahead and implement this, makes it a lot more readable ty